_studio/shared/umc/codec/h264_dec/include/umc_h264_task_supplier.h

```diff
@@ -775,6 +775,8 @@ inline int32_t CalculateDPBSize(uint8_t & level_idc, int32_t width, int32_t heig
         case H264VideoDecoderParams::H264_LEVEL_52:
             level_idc = H264VideoDecoderParams::H264_LEVEL_MAX;
             break;
+        default:
+            throw h264_exception(UMC_ERR_INVALID_PARAMS);
```

Comment:
@dmitry-gurulev : if this function CalculateDPBSize will be called with the invalid level value _unknown_ to this switch, we could enter infinite loop. I guess that's what @aulitin was trying to say. If I understand author's commit message correctly, level_idc was initially passed as =240. Question @aulitin : where we managed to take this value from?

So, the above switch is designed to fix/adjust _correct_ value of level_ids only. If it will be wrong we will blow up.

Reply:
@dvrogozh 
>if this function CalculateDPBSize will be called with the invalid level

My point is that it shouldn't be called in that way and we need to fix initial problem - appearing wrong level value. The second thing is it's a logic error and exception should be `UMC_ERR_FAILED`, not `UMC_ERR_INVALID_PARAMS`

Reply:
@dmitry-gurulev : my point is the same. I completely agree. This patch attempt to fix only the second part. The first part with wrong level value is not fixed and probably is not fully understood.